File: leetcode-solutions/0005-longest-palindromic-substring/solution.c
int ispall(int i,int j,char str[]){
    while(i<=j){
        if(str[i]!=str[j]){
            return 0;
        }
        i++;
        j--;
    }
    return 1;
    }
char* longestPalindrome(char* str) {
   static char str1[10000]={0};
    int len=strlen(str);
    int max=0,start,k=0;
    for(int i=0;i<len;i++){
        for(int j=i;j<len;j++){
            if(ispall(i,j,str)==1){
                int c=j-i+1;
                if(c>max){
                    max=c;
                    start=i;
                }
            }
        }
    }
    for(int i=start;i<start+max;i++){
        str1[k++]=str[i];
    }
    str1[k]='\0';
    return str1;
}

File: leetcode-solutions/0005-longest-palindromic-substring/solution.java
class Solution {
    public String longestPalindrome(String s) {
        int start=0,end=0;
        for(int i=0;i<s.length();i++){
            int left=i;
            int right=i;
            while(left>=0&&right<s.length()&&s.charAt(left)==s.charAt(right)){
                left--;
                right++;
            }
            int len1=right-left-1;
            left=i;
            right=i+1;
            while(left>=0&&right<s.length()&&s.charAt(left)==s.charAt(right)){
                left--;
                right++;
            }
            int len2=right-left-1;
            int maxLen=Math.max(len1,len2);
            if(maxLen>end-start){
                start=i-(maxLen-1)/2;
                end=i+maxLen/2;
            }
        }
        return s.substring(start,end+1);
    }
}
